contextily/plotting.py

```diff
@@ -222,6 +226,44 @@ def _reproj_bb(left, right, bottom, top, s_crs, t_crs):
     n_l, n_b, n_r, n_t = transform_bounds(s_crs, t_crs, left, bottom, right, top)
     return n_l, n_r, n_b, n_t
 
+def _is_overlay(url):
+    if not isinstance(url, dict):
+        return False
+    if url.get('opacity', 1.0) < 1.0:
+        return True
+    overlayPatterns = [
+        '^(OpenWeatherMap|OpenSeaMap)',
+        'OpenMapSurfer.(Hybrid|AdminBounds|ContourLines|Hillshade|ElementsAtRisk)',
+        'Stamen.Toner(Hybrid|Lines|Labels)',
+        'Hydda.RoadsAndLabels',
+        '^JusticeMap',
+        'OpenPtMap',
+        'OpenRailwayMap',
+        'OpenFireMap',
+        'SafeCast'
+    ]
+    import re
+    return bool(re.match('(' + '|'.join(overlayPatterns) + ')', url.get('name','')))
+
+
+# if (layer.options.opacity && layer.options.opacity < 1) {
```

Comment:
Can you add a comment and link to the original source to explain this part?

Reply:
Link back to the original leaflet.js version in docstring for this function.